homeassistant/components/aidot/light.py

```diff
@@ -0,0 +1,273 @@
+
+    @callback
+    def _async_check_devices():
+        filter_cloud_devices = coordinator.filter_light_list()
+        device_registry: DeviceRegistry = dr.async_get(hass)
+        cloud_device_ids = [device[CONF_ID] for device in filter_cloud_devices]
+        ha_exist_device_ids = [
+            identifier[1]
+            for device_entry in device_registry.devices.values()
+            for identifier in device_entry.identifiers
+            if identifier[0] == DOMAIN and entry.entry_id in device_entry.config_entries
+        ]
+
+        # add new device
+        new_devices = [
+            device
+            for device in filter_cloud_devices
+            if device[CONF_ID] not in ha_exist_device_ids
+        ]
+        if new_devices:
+            async_add_entities(
+                AidotLight(hass, device_info, coordinator)
+                for device_info in new_devices
+                if device_info[CONF_TYPE] == Platform.LIGHT
+                and CONF_AES_KEY in device_info
+                and device_info[CONF_AES_KEY][0] is not None
+            )
+
+        # remove device
+        removed_device_ids = set(ha_exist_device_ids) - set(cloud_device_ids)
+        for removed_device_id in removed_device_ids or []:
+            if device := device_registry.async_get_device(
+                identifiers={(DOMAIN, removed_device_id)}
+            ):
+                device_registry.async_remove_device(device.id)
+
+    coordinator.async_add_listener(_async_check_devices)
+    async_add_entities(
+        AidotLight(hass, device_info, coordinator)
+        for device_info in coordinator.filter_light_list()
+    )
+
+
+class AidotLight(CoordinatorEntity, LightEntity):
+    """Representation of a Aidot Wi-Fi Light."""
+
+    async def async_added_to_hass(self) -> None:
+        """Connect to device when this entity is added to HA."""
+        await super().async_added_to_hass()
+        await self._try_connect()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Cleanup resources when entity will be removed from hass."""
+        await super().async_will_remove_from_hass()
+        await self._cleanup_resources()
+
+    def __init__(
+        self, hass: HomeAssistant, device, coordinator: AidotCoordinator
+    ) -> None:
+        """Initialize the light."""
+        super().__init__(coordinator, {device[CONF_ID]})
+        self.device = device
+        self.user_info = coordinator.client.login_info
+        self._attr_unique_id = device[CONF_ID]
+        self._attr_name = device[CONF_NAME]
+        self.pingtask = None
+        self.recvtask = None
+
+        modelId = device[CONF_MODEL_ID]
+        manufacturer = modelId.split(".")[0]
+        model = modelId[len(manufacturer) + 1 :]
+        mac = format_mac(device[CONF_MAC]) if device[CONF_MAC] is not None else ""
+        identifiers: set[tuple[str, str]] = (
+            set({(DOMAIN, self._attr_unique_id)}) if self._attr_unique_id else set()
```

Comment:
```suggestion
            set({(DOMAIN, self._attr_unique_id)})
```
unique id is always set